WordPress/Classes/Models/Blog/Blog.m

```diff
@@ -97,6 +97,17 @@ @implementation Blog
 
 #pragma mark - NSManagedObject subclass methods
 
+- (void)willSave {
+    [super willSave];
+
+    // The `dotComID` getter has a speicial code to _update_ `blogID` value.
+    // This is a weird patch to make sure `blogID` is set to a correct value.
+    //
+    // It's important that calling `[self dotComID]` repeatedly only updates
+    // `Blog` instance once, which is the case at the moment.
+    [self dotComID];
```

Comment:
It looks like a good place to do it: "This method can have “side effects” on persistent values. You can use it to, for example, compute persistent values from other transient or scratchpad values." 👍 

Having said that, this rename to `dotComID` has been a real PITA. It's be nice to do a migration and remove the original field. I plan to wrap up this https://github.com/wordpress-mobile/WordPress-iOS/pull/23788 this week and add more stuff that can be removed form the database. And there are a ton of existing deprecations, especially in `Blog`.